src/include/daos/common.h

```diff
@@ -495,6 +496,11 @@ void daos_iov_append(d_iov_t *iov, void *buf, uint64_t buf_len);
 static inline int
 daos_errno2der(int err)
 {
+	if (err < 0) {
+		D_ERROR("error < 0 (%d)\n", err);
+		return -DER_UNKNOWN;
+	}
+
```

Comment:
With these two checks I did wonder about leaving them out of release builds as it's basically a sanity check.